crates/cli/src/subcommands/generate/csharp.rs

```diff
@@ -407,28 +407,28 @@ fn autogen_csharp_access_funcs_for_struct(
             None => continue,
             Some(x) => x,
         };
-        writeln!(
-            output,
-            "public readonly ref struct {csharp_field_name_pascal}UniqueIndex"
-        );
+        writeln!(output, "public class {csharp_field_name_pascal}UniqueIndex");
         indented_block(output, |output| {
+            write!(
+                output,
+                "internal readonly Dictionary<{csharp_field_type}, {csharp_table_name}> __Cache = new(16);"
```

Comment:
Why do we need `__` - can it clash with anything provided by the user? I don't think so because we don't add any user-generated names in this class, do we?